client/settings/general-settings/index.js

```diff
@@ -133,6 +143,17 @@ const GeneralSettings = () => {
 					closeModal={ () => setModalVisible( false ) }
 				/>
 			) }
+			{ devConfirmationModalVisible && (
+				<DevModeConfirmationModal
+					onClose={ () => {
+						updateIsTestModeEnabled( false );
+						setDevConfirmationModalVisible( false );
+					} }
+					onConfirm={ () => {
+						setDevConfirmationModalVisible( false );
+					} }
```

Comment:
Similar another suggestion for `CheckBoxControl`, my suggestion is to reflect our intention behind this modal:

```suggestion
					onClose={ () => {
						setDevConfirmationModalVisible( false );
					} }
					onConfirm={ () => {
						updateIsTestModeEnabled( true );
						setDevConfirmationModalVisible( false );
					} }
```

Reply:
Done